76 double click card bug

pontaberglund opened the pull request:
### Tasks complete
- Add animation state to model
- Check animation state in all <Link> components
- Reformat browsePresenter (A.K.A Ctrl + s)

### Testing
1. The bug  #76 was that if you spam on a character from browse view the URL broke. This is now fixed.
2. Another problem I noticed was that when you spam on a card, the history gets updated for every click. If you managed to click a card 3 times during the animation, you could press the back arrow 3 times. This should also be fixed.

### Known issues
If you spam from the landing page when the characters are cached. So you spam click characters and then directly spam for example Hyper Rod, the history will be duplicated.

pilsnerfrajz reviewed (changes requested):
The fix does not work in the profile view. If you click multiple times on a card, the URL changes to `/profile/undefined/Yoda`.

pontaberglund commented:
> The fix does not work in the profile view. If you click multiple times on a card, the URL changes to `/profile/undefined/Yoda`.

Should be fixed now. The known issue is easier to demonstrate on Discord. Will do a rebase after Friends have been merged since I changed how <Link> works and added some props to BrowseView.

pilsnerfrajz reviewed (approved):
Bug seems fixed. LGTM.
